### ISSUE TITLE
Using variable labels instead of variable names when available

### ISSUE THREAD

**Issue by larmarange** [NikNakk/forestmodel#24]
Variable labels, stored as a `label` attributes and easily accessible with `labelled::var_label()`, are becoming quite common. Many packages (like `gtsummary`) producing graphs or tables are now adopting the following rule: if defined, use variable labels instead of variable names.

Such addition to `forestmodel` would allow to easily customize the names of variables displayed on forest plots.

**Comment by ShixiangWang** [comment]
This package is not in active development, if you are interested in this feature, please implement it, then keep a fork or create a pull request to https://github.com/ShixiangWang/forestmodel

**Comment by larmarange** [comment]
@ShixiangWang is it an official fork?

@NikNakk could you clarify if you still plan to maintain and develop `forestmodel`?

**Comment by ShixiangWang** [comment]
@larmarange Nope, I don't say that. The author is nice, but he may be not active in GitHub, from my view.

**Comment by NikNakk** [comment]
Hi @larmarange, @ShixiangWang,

I've not been very active in maintaining this package for a while because of being busy with other things, but I'm still aiming to get to the outstanding queries that have been raised including yours. There's also now a more pressing reason to attend to the package because it's erroring on CRAN so will be delisted if I don't fix that. I'll at least fix the current issue that would lead to delisting in the next few days, but if I can I'll try to fix any other outstanding issues and improvements.

**Comment by larmarange** [comment]
Thanks @NikNakk for your feedback.

Regarding the proposed improvement, it should not be very difficult to implement once identified where variable names are taken into account. 

I didn't have time to get into your code in details so I do not know yet how your code was organized. But as you are familiar with your package, you should have an idea on where to look at.

Best regards

**Comment by NikNakk** [comment]
@larmarange I've made a new branch that has a simple implementation of this at https://github.com/NikNakk/forestmodel/tree/labels. You can test it using remotes::install_github("NikNakk/forestmodel@labels")

**Comment by larmarange** [comment]
Thanks a lot

**Comment by NikNakk** [comment]
@larmarange please let me know when you've had a chance to test this out.

**Comment by larmarange** [comment]
@NikNakk I have done some quick tests. It works well with simple models. Thanks.

When I add interaction terms, labels are not taken into account for interaction terms, but it was already the case before (it seems that `forstmodel` was not treating them in a particular way).

```r
library(questionr)
library(forestmodel)
library(labelled)

data(fertility)
women <- unlabelled(women)
mod <- glm(employed ~ age + residency * instruction, data = women, family = binomial())
forest_model(mod, exponentiate = TRUE)
```
![image](https://user-images.githubusercontent.com/966307/87802026-dda8fd80-c850-11ea-8469-e8abf9d6a5b0.png)


Here a quick example with `gtsummary` to show this package handle interaction terams.

```
library(gtsummary)
tbl_regression(mod)
```

|**Characteristic**                             |**log(OR)** |**95% CI**  |**p-value** |
|:----------------------------------------------|:-----------|:-----------|:-----------|
|Age at last anniversary (in years)             |0.06        |0.05, 0.07  |<0.001      |
|Urban / rural residency                        |            |            |            |
|urban                                          |            |            |            |
|rural                                          |0.28        |0.00, 0.55  |0.052       |
|Level of instruction                           |            |            |            |
|none                                           |            |            |            |
|primary                                        |0.35        |-0.02, 0.74 |0.067       |
|secondary                                      |-0.83       |-1.2, -0.50 |<0.001      |
|higher                                         |-0.71       |-1.3, -0.10 |0.022       |
|Urban / rural residency * Level of instruction |            |            |            |
|rural * primary                                |-0.16       |-0.67, 0.35 |0.5         |
|rural * secondary                              |0.19        |-0.41, 0.80 |0.5         |
|rural * higher                                 |-1.5        |-4.5, 0.56  |0.2         |

**Comment by larmarange** [comment]
But I know that managing interaction terms could be tricky and beyond the current issue.

Otherwise, it's perfect. Thanks a lot

**Comment by NikNakk** [comment]
I’ll have a look at interaction terms when I get a chance. `gtsummary` looks like a good starting point. For now I’ve merged the labels branch into master and need to get the latest version on CRAN because otherwise it will be delisted.

**Comment by larmarange** [comment]
Thanks

**Comment by NikNakk** [comment]
FYI, this version is now on CRAN.

**Comment by proshano** [comment]
Variable labels still not showing up

**Comment by corneliushennch** [comment]
> Variable labels still not showing up

Same here, it works fine with `gtsummary::tbl_regression` but not with `forest_model` from the `forestmodel` package that I just downloaded from Github.

**Comment by NikNakk** [comment]
Sorry for the delayed response, @proshano and @corneliushennch. Could you please give me some example code that doesn't work as expected? I'm still planning to work on interaction terms since they're not currently properly supported with or without labels.

**Comment by larmarange** [comment]
In case it could be useful for you, `gtsummary::tbl_regression()` now relies on `broom.helpers` package: https://larmarange.github.io/broom.helpers/

**Comment by corneliushennch** [comment]
I actually just figured it out: The problem occurs with factor variables (which are super useful to use in this kind of data analysis, as other packages e.g. `tableone` need factors...). I made (my very first) `reprex` to highlight the problem:

``` r
library(survival)
library(dplyr)
#> 
#> Attaching package: 'dplyr'
#> The following objects are masked from 'package:stats':
#> 
#>     filter, lag
#> The following objects are masked from 'package:base':
#> 
#>     intersect, setdiff, setequal, union
library(forestmodel)
#> Loading required package: ggplot2

surv_data <- tibble(
  time = abs(rnorm(300, 50, 30)),
  event = sample(c(0,1), 300, prob = c(0.8, 0.2), replace = TRUE),
  gender = sample(c(0,1), 300, prob = c(0.6, 0.4), replace = TRUE),
  rx = sample(c(0,1), 300, prob = c(0.5, 0.5), replace = TRUE),
  gene = sample(c(0,1), 300, prob = c(0.9, 0.1), replace = TRUE)
)

surv_data <- surv_data %>% 
  mutate(gender = factor(gender, levels = c(0,1), labels = c("female", "male")))

labelled::var_label(surv_data) <- list(
  gender = "Gender (f/m)", #this variable is a factor -> doesn't work!
  rx = "Irradiation", # not a facotr -> label works!
  gene = "Gene of Interest" # not a factor -> label works!
)

labelled::var_label(surv_data) # labels are assigned
#> $time
#> NULL
#> 
#> $event
#> NULL
#> 
#> $gender
#> [1] "Gender (f/m)"
#> 
#> $rx
#> [1] "Irradiation"
#> 
#> $gene
#> [1] "Gene of Interest"

print(forest_model(coxph(formula = Surv(time, event) ~ gender + rx + 
                           gene, data = surv_data)))
#> Warning in recalculate_width_panels(panel_positions, mapped_text =
#> mapped_text, : Unable to resize forest panel to be smaller than its heading;
#> consider a smaller text size
```

![](https://i.imgur.com/8qIgjyj.png)

``` r
# printing variable class -> the factor doesn't work...
lapply(surv_data, class)
#> $time
#> [1] "numeric"
#> 
#> $event
#> [1] "numeric"
#> 
#> $gender
#> [1] "factor"
#> 
#> $rx
#> [1] "numeric"
#> 
#> $gene
#> [1] "numeric"
```

<sup>Created on 2021-04-22 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by fabones1** [comment]
I would also love to have the coxph variable labels bug fixed as it would save a lot of time in my work.

**Comment by mhmdrahouma** [comment]
Thanks a lot for all your efforts. Please let us know if the label bug got fixed for `coxph` as it would save us a lot of time. I tried it but it didn't work. It works perfectly with `glm` only. Appreciate your help and advice.
